### DISCUSSION TITLE
Making it usable for ZeroNet

### DISCUSSION THREAD

**Discussion by mahdi-ln** [arvidn/libtorrent#6881] [Ideas]
I wish you make libtorrent DHT, TCP compatible so ZeroNet developers would use it. It need something compatible with Tor. 
ZeroNet is only decentralized web project that haven't acted as brake and really provided a usable client
Website of a fork with activity:  https://zeronet.dev/

**Comment by arvidn** [comment]
You mean you want the DHT to run over TCP connections?

**Comment by mahdi-ln** [comment]
Yes

**Reply by arvidn** [reply]
the DHT is carefully made to not require handshakes. The only 3-way, anti-spoofing, message is when announcing your IP. You first need to have made a lookup to get the "syn-cookie" (or write_token as it's called in DHT).

establishing a TCP connection for every (or the majority of) DHT messages being sent and received would incur quite a massive additional load. Is this something that has been thought through?
